Sign stack usage: Re-use y/h buffer

mkannwischer opened the pull request:
This commit is the second commit working towards bringing down the memory consumption of signature_internal. It combines the y and h buffer as those lifetime does not overlap.

- Hoisted out from https://github.com/pq-code-package/mldsa-native/pull/791

`-fstack-usage` of `signature_internal` of ML-DSA-86 reduces from 121536 to 114368 (aarch64, -Os, gcc14).

mkannwischer committed:
mld_attempt_signature_generation: Introduce wrapper for allocations

Signed-off-by: Matthias J. Kannwischer <matthias@kannwischer.eu>

hanno-becker committed:
[WIP] Reduce CI

Signed-off-by: Hanno Becker <beckphan@amazon.co.uk>

hanno-becker commented:
@mkannwischer To double-check: Can you try adding a helper function which takes the pointers to the helper structures (incl. `y` and `h`) as arguments, and simply asserting that `y` and `h` be the same? I wonder if that would lead to a different handling of read/writes than when `y` and `h` are visibly (to CBMC) members of a union.

mkannwischer committed:
sign stack usage: Re-use y/h buffer

This commit is the second commit working towards bringing down the memory
consumption of signature_internal. It combines the y and h buffer as those
lifetime does not overlap.

Signed-off-by: Matthias J. Kannwischer <matthias@kannwischer.eu>

hanno-becker committed:
[TEST] Reduce visual clutter in CBMC specs

Signed-off-by: Hanno Becker <beckphan@amazon.co.uk>

mkannwischer commented:
We merged #818  instead.
